src/com/projectkorra/projectkorra/waterbending/OctopusForm.java

```diff
@@ -239,105 +239,98 @@ public void progress() {
+				} else {
+					this.remove();
+				}
+			} else if (this.sourceBlock.getY() > location.getBlockY()) {
+				this.source.revertBlock();
+				this.source = null;
+				final Block newBlock = this.sourceBlock.getRelative(BlockFace.DOWN);
+				this.sourceLocation = newBlock.getLocation();
+
+				if (!GeneralMethods.isSolid(newBlock)) {
+					this.source = new TempBlock(newBlock, GeneralMethods.getWaterData(0));
+					this.sourceBlock = newBlock;
+				} else {
+					this.remove();
+				}
+			} else if (this.sourceLocation.distanceSquared(location) > this.radius * this.radius) {
+				final Vector vector = GeneralMethods.getDirection(this.sourceLocation, location.getBlock().getLocation()).normalize();
+				this.sourceLocation.add(vector);
+				final Block newBlock = this.sourceLocation.getBlock();
 
-					if (!GeneralMethods.isSolid(newBlock)) {
-						this.source = new TempBlock(newBlock, GeneralMethods.getWaterData(0));
-						this.sourceBlock = newBlock;
-					} else {
-						this.remove();
-						return;
+				if (!newBlock.equals(this.sourceBlock)) {
+					if (this.source != null) {
+						this.source.revertBlock();
 					}
-				} else if (this.sourceBlock.getY() > location.getBlockY()) {
-					this.source.revertBlock();
-					this.source = null;
-					final Block newBlock = this.sourceBlock.getRelative(BlockFace.DOWN);
-					this.sourceLocation = newBlock.getLocation();
-
 					if (!GeneralMethods.isSolid(newBlock)) {
 						this.source = new TempBlock(newBlock, GeneralMethods.getWaterData(0));
 						this.sourceBlock = newBlock;
-					} else {
-						this.remove();
-						return;
 					}
-				} else if (this.sourceLocation.distanceSquared(location) > this.radius * this.radius) {
-					final Vector vector = GeneralMethods.getDirection(this.sourceLocation, location.getBlock().getLocation()).normalize();
-					this.sourceLocation.add(vector);
-					final Block newBlock = this.sourceLocation.getBlock();
-
-					if (!newBlock.equals(this.sourceBlock)) {
-						if (this.source != null) {
-							this.source.revertBlock();
-						}
-						if (!GeneralMethods.isSolid(newBlock)) {
-							this.source = new TempBlock(newBlock, GeneralMethods.getWaterData(0));
-							this.sourceBlock = newBlock;
-						}
-					}
-				} else {
-					this.incrementStep();
-					if (this.source != null) {
-						this.source.revertBlock();
-					}
-
-					this.source = null;
-					final Vector vector = new Vector(1, 0, 0);
-					this.startAngle = vector.angle(GeneralMethods.getDirection(this.sourceBlock.getLocation(), location));
-					this.angle = this.startAngle;
 				}
-			} else if (this.forming) {
-				if (this.angle - this.startAngle >= 360) {
-					this.currentFormHeight += 1;
-				} else {
-					this.angle += 20;
+			} else {
+				this.incrementStep();
+				if (this.source != null) {
+					this.source.revertBlock();
 				}
 
-				if (random.nextInt(4) == 0) {
-					playWaterbendingSound(this.player.getLocation());
-				}
+				this.source = null;

```

Comment:
Looks good. But can you confirm it's tested? Just want to make sure Octopus works as intended and no side effects occur from refactoring that we can't see

Reply:
Tested on 1.16.5 with version compiled from source code.

![image](https://user-images.githubusercontent.com/44327892/194733669-2b28a04f-2990-495e-8181-bb064da31804.png)

Actually code is same, almost nothing changed.
* new created Random instance is moved to inside of statement. Because it doesn't used in all statements.
* An if statement is inverted.

Reply:
This is not an ideal change, there shouldn't be any issue with reusing one instance of Random verses creating a new one each time you want to check a number. That being said, a better change would be to use [ThreadLocalRandom](https://docs.oracle.com/javase/8/docs/api/java/util/concurrent/ThreadLocalRandom.html).